sites/topomath.asu.edu/modules/system_descriptions/static/sys_desc.js

```diff
@@ -0,0 +1,261 @@
+									// calculate the percentage of upload completed
+									var percentComplete = evt.loaded / evt.total;
+									percentComplete = parseInt(percentComplete * 100);
+									// update the Bootstrap progress bar with the new percentage
+									$('.progress-bar').show();
+									$('.progress-bar').text(percentComplete + '%');
+									$('.progress-bar').width(percentComplete + '%');
+									// once the upload reaches 100%, set the progress bar text to done
+									if (percentComplete === 100) {
+										$('.progress-bar').html('Done');
+										//$('.progress-bar').hide();
+										$('#list-sys-descriptions').append("<div><a style='cursor: pointer' href='"+uploaddir+pathName+"' target='_blank' >"+fileName+"</a><span class='upload_glyph glyphicon glyphicon-remove delete-sys-desc-inline' data-syspath='"+pathName+"''></span></div><br/>");
+										//this should also update the database in this case
+										console.log(formData);
+									}
+								}
+							}, false);
+							return xhr;
+						}
+					});
+					//add new description updates the database with the currently uploaded new description
+					addNewDescription(fileName,pathName);
+				}
+				else{
+					$('#sysDesc-error-box').html('File with given name already exists');
+				}
+			});
+		}
+	}
+	});
+
+	//loading system descriptions inside
+
+
+	$('#add-desc-text').on('click',function(){
+		var form = document.forms['dragoon_problem_form'];
+		var user = form["u"].value;
+		var prob_name = form["p"].value;
+		var folder = form["f"].value;
+		var sec = form["s"].value;
+		$('#handleSystemDesc').modal('hide');
+		window.open(cke_path+'?p='+prob_name+'&f='+folder+'&s='+sec+'&u='+user, '_blank');
+	});
+
+
+	/* updateDescList : updates the system description list by sending a query to db
+	* one use of this function in updating the list of system descriptions once a new description is created via ck editor
+	* uploading system descriptions updates the list without the help of this function
+	*/
+	var updateDescList = function(){
+		//fill the input data structure 
+		//inputs are user name, folder and section
+		//step 1: collect data relevant to the operations of this module
+		$('#list-sys-descriptions').html('');
+		var form = document.forms['dragoon_problem_form'];
+		var folder = form["f"].value;
+		var problem = form["p"].value;
+		var section = form["s"].value;
+		var input = {
+			f: folder,
+			p: problem,
+			s: section,
+			req_type: "refreshList"
+		}
+		$.post(sd_update_handler, input)
+			.success(function (datum) {
+				console.log("received",datum);
+				var sys_json = JSON.parse(datum);
+				for(var sdName in sys_json){
+					console.log(sdName, sys_json[sdName]["path"], sys_json[sdName]["type"]);
+					var descName = sdName.trim();
+					console.log("name", descName);
+					if(descName!= ""){
+						var descpath = sys_json[sdName]["path"].trim();
+						var descLink = uploaddir+descpath;
+						var descType = sys_json[sdName]["type"].trim();
```

Comment:
all variable naming schemes are again used here. How about using a only one of them and also for function names. Thanks